lib/nautic_net_web/live/sailboat_polars_live.ex

```diff
@@ -50,24 +50,29 @@ defmodule NauticNetWeb.SailboatPolarsLive do
     ]
   end
 
-  def mount(_, _, socket) do
-    {polars_df, run_beat_df} =
-      SailboatPolars.load(Path.join(:code.priv_dir(:nautic_net), "sample_sailboat_csv.csv"))
-
-    sku = "h-2-2023-11133-7929-0-579"
+  def mount(params, _, socket) do
+    sku = params["sku"] || ""
+    sail_kind = params["sail_kind"] || "spin"
 
     socket =
-      socket
-      |> assign(
+      assign(socket,
         csv_changeset: csv_changeset(%{}, []),
         rendered_csv: "",
         export_csv: %{},
-        sail_kind: "spin",
+        sail_kind: sail_kind,
         desired_angles: Enum.to_list(20..180//10),
         sku: sku,
         plot_title: "Boat Speed (kts) [#{sku} - Spinnaker]"
       )
-      |> load(polars_df, run_beat_df)
+
+    socket =
+      if sku != "" do
+        {polars_df, run_beat_df} = SailboatPolars.load_from_url(sku, sail_kind)
+
+        load(socket, polars_df, run_beat_df)
+      else
+        assign(socket, sailboat_polars: [], run_beat_polars: [], interpolation_by_tws: [])
+      end
```

Comment:
@murtza Ideally this would be done after mounting so that the page first loads in an empty state. Any tips on how I can do this?

Reply:
Most of the data is loading in `def select_date/1` function. You can put this in there.